src/lib/wallets/useIsContractAccount.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from "react";
+import { useWeb3React } from "@web3-react/core";
+
+export default function useIsContractAccount() {
+  const { active, account, library } = useWeb3React();
+  const [isContract, setIsContract] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    if (!active) return;
+    (async function () {
+      const code = await library.getCode(account);
```

Comment:
should we memoize it? EOA address can't become contract and vice versa. so no need to send request again 

or maybe i'm over-optimizing, there are a lot of non-optimized things anyway :)